orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/DefaultVariableAssignmentTransform.java

```diff
@@ -59,6 +60,36 @@ public void visitPipelineTemplate(PipelineTemplate pipelineTemplate) {
       throw new IllegalTemplateConfigurationException("Missing variable values for: " + StringUtils.join(missingVariables, ", "));
     }
 
-    // TODO rz - validate variable values match the defined variable type
+    // collect variables where value type doesn't match the required type
+    List<String> wrongTypeErrorMessages = pipelineTemplateVariables.stream()
+      .filter(templateVar -> {
+        if (missingVariables.contains(templateVar.getName())) {
+          return false; // filter missing values from this list
+        }
+
+        String expectedType = templateVar.getType();
+        if (expectedType.equals("object")) {
+          return false; // not invalid, all classes are objects
+        }
+
+        String actualType = configVars.get(templateVar.getName()).getClass().getSimpleName();
+
+        if (expectedType.equals("int") && actualType.equals("Integer")){
+          return false;
+        } else if (expectedType.equals("list") && actualType.equals("ArrayList")) {
+          return false;
+        } else if (expectedType.equals(actualType.toLowerCase())) {
+          return false;
+        } else if (expectedType.equals(actualType)){
+          return false; // types match
+        }
```

Comment:
Lets change these all to `equalsIgnoreCase`, then we can get rid of the last conditional.

Further, I'm kinda wondering if matching against the actual type might be a better way forward, so we don't run into any funny business (esp around lists).

```java
String expectedType = templateVar.getType();
if (expectedType.equalsIgnoreCase("object") {
  return false;
}

Class<?> actualType = configVars.get(templateVar.getName()).getClass();

return !((expectedType.equalsIgnoreCase("int") && actualType instanceof Integer.class) ||
  (expectedType.equalsIgnoreCase("bool") && actualType instanceof Boolean) ||
  (expectedType.equalsIgnoreCase("list") && actualType instanceof Collection) ||
  (expectedType.equalsIgnoreCase("string") && actualType instanceof CharSequence.class) ||
  (expectedType.equalsIgnoreCase(actualType.getSimpleName()));
```

Reply:
Changing.